gemrb/core/Scriptable/Actor.cpp

```diff
@@ -1131,7 +1131,7 @@ static void pcf_class (Actor *actor, ieDword /*oldValue*/, ieDword newValue)
 {
 	//Call forced initbuttons in old style systems, and soft initbuttons
 	//in case of iwd2. Maybe we need a custom quickslots flag here.
-	actor->InitButtons(newValue, !iwd2class);
+	actor->InitButtons(actor->GetActiveClass(), !iwd2class);
```

Comment:
if i remember testing correctly, this is not how the original behaved. Do you have any data to the contrary?

Reply:
I've just re-tested this in BG2 (a Trilogy build with ToBEx, to be specific) by creating a thief PC and dualing him into mage.  After dual he got spell quickslots instead of thief ability buttons, only regaining the latter after thief class reactivation.  It might possibly be a ToBEx thing, though.

Reply:
Ok, I'll recheck on vanilla.